src/fiatconnect/BankDetailsScreen.tsx

```diff
@@ -0,0 +1,275 @@
+import { FiatAccountSchema, FiatConnectError } from '@fiatconnect/fiatconnect-types'
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useLayoutEffect, useMemo, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Image, ScrollView, StyleSheet, Text, View } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useDispatch } from 'react-redux'
+import { showError, showMessage } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import BorderlessButton from 'src/components/BorderlessButton'
+import Button, { BtnSizes } from 'src/components/Button'
+import TextInput from 'src/components/TextInput'
+import { addNewFiatAccount } from 'src/fiatconnect'
+import i18n from 'src/i18n'
+import ForwardChevron from 'src/icons/ForwardChevron'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'BankDetailsScreen'
+type ScreenProps = StackScreenProps<StackParamList, Screens.BankDetailsScreen>
+
+type Props = ScreenProps
+
+interface Fields {
+  name: string
+  label: string
+  regex: RegExp
+  placeholderText: string
+  errorMessage: string
+}
+
+// This is a mapping between different fiat account schema to the metadata of the fields that need to be rendered on the bank details screen
+const SCHEMA_TO_FIELD_METADATA_MAP = {
+  AccountNumber: [
+    {
+      name: 'accountName',
+      label: i18n.t('fiatAccountSchema.accountName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
+    },
+    {
+      name: 'institutionName',
+      label: i18n.t('fiatAccountSchema.institutionName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
+    },
+    {
+      name: 'accountNumber',
+      label: i18n.t('fiatAccountSchema.accountNumber.label'),
+      regex: /^\d{10}$/,
```

Comment:
```suggestion
      regex: /^\[0-9]{10}$/,
```
We don't want to include non-latin digit characters. https://stackoverflow.com/a/273144/18163451